### ISSUE TITLE
RGB image export from Plot2D with original resolution

### ISSUE THREAD

**Issue by linupi** [silx-kit/silx#437]
On the one hand the ability of silx/Plot2D to directly export greyscale tiffs from the Plot2D window is great for further data processing and on the other the export of png and jpg snapshots from the current graph is handy for electronic logbooks etc. 

It would be great to also have the possibility to export a false color image in the resolution of the original data without any axes or labels (just using the colormap as used in the plot window applied to the data as exported in the tiff/edf export). This would be great to have for images that one wants to use in publications and would give the option to later on add separate overlay layers to the exported image as the experimental resolution of the raster-data is usually known.

**Comment by vasole** [comment]
I guess you mean to export it from the GUI.

Probably the method to_rgba in http://matplotlib.org/api/cm_api.html can do the job.

**Comment by t20100** [comment]
This can be added to the supported outputs.
Yes, internally, it means giving access to matplotlib through the silx API up to the save dialog.

**Comment by linupi** [comment]
Yes, intentionally I meant an export from the GUI and I still think that would be the best.
However, I have played around with to_rgba() from matplotlib but I could not figure out how to use the silx colormaps dictionary that I can get via getDefaultColormap() from a Plot2D within matplotlib.cm as cmap (e.g. in combinatation with ScalarMappable.set_cmap() which I would need in order to use to_rgba() ).

**Comment by t20100** [comment]
silx plot colormap description is different from matplotlib.

Here is a sample code to convert silx colormap (all but the 'name') to a matplotlib ScalarMappable:

```
import numpy

from matplotlib.colors import LogNorm, Normalize
from matplotlib import cm

silx_colormap = {
    'name': 'gray',
    'normalization': 'linear',
    'autoscale': True,
    'vmin': 0.0,
    'vmax': 1.0}

# Conversion to matplotlib normalization
if silx_colormap['autoscale']:
    vmin, vmax = None, None
else:
    vmin, vmax = silx_colormap['vmin'], silx_colormap['vmax']

if silx_colormap['normalization'] == 'linear':
    norm = Normalize(vmin, vmax)
else:
    norm = LogNorm(vmin, vmax)

# Converting the name is not direct
# Some silx plot colormaps are defined internally:
# See https://github.com/silx-kit/silx/blob/master/silx/gui/plot/BackendMatplotlib.py#L765
# There is equivalent colormap in matplolib, see http://matplotlib.org/users/colormaps.html
cmap = cm.gray

scalar_mappable = cm.ScalarMappable(norm, cmap)

# 2D data to convert to RGBA
data = numpy.arange(100).reshape(10, 10)

rgba_image = scalar_mappable.to_rgba(data)
```

**Comment by t20100** [comment]
Addressed in PR #526, see remark in PR https://github.com/silx-kit/silx/pull/526#issuecomment-272869532
